crates/katana-rpc/src/starknet/rpc_test.rs

```diff
@@ -0,0 +1,91 @@
+use std::fs::{self, File};
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet::contract_class::ContractClass;
+use dojo_test_utils::sequencer::TestSequencer;
+use starknet::accounts::{Account, ConnectedAccount};
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::contract::{CompiledClass, SierraClass};
+use starknet::core::types::{
+    DeclareTransactionReceipt, FieldElement, FlattenedSierraClass, MaybePendingTransactionReceipt,
+    TransactionReceipt, TransactionStatus,
+};
+use starknet::providers::Provider;
+
+#[tokio::test]
+async fn test_send_declare_v2_tx() {
+    let sequencer = TestSequencer::start().await;
+    let account = sequencer.account();
+
+    let path: PathBuf = PathBuf::from("src/starknet/test_data/cairo1_contract.json");
+    let (contract, class_hash) = prepare_contract_declaration_params(&path).unwrap();
+
+    let res = account.declare(Arc::new(contract), class_hash).send().await.unwrap();
+    let receipt = account.provider().get_transaction_receipt(res.transaction_hash).await.unwrap();
+
+    sequencer.stop().expect("failed to stop sequencer");
+
+    match receipt {
+        MaybePendingTransactionReceipt::Receipt(TransactionReceipt::Declare(
+            DeclareTransactionReceipt { status, .. },
+        )) => {
+            assert_eq!(status, TransactionStatus::AcceptedOnL2);
+        }
+        _ => panic!("invalid tx receipt"),
+    }
+}
+
+#[tokio::test]
+async fn test_send_declare_v1_tx() {
+    let sequencer = TestSequencer::start().await;
+    let account = sequencer.account();
+
+    let path = PathBuf::from("src/starknet/test_data/cairo0_contract.json");
+
+    let legacy_contract: LegacyContractClass =
+        serde_json::from_reader(fs::File::open(path).unwrap()).unwrap();
+    let contract_class = Arc::new(legacy_contract);
+
+    let res = account.declare_legacy(contract_class).send().await.unwrap();
+    let receipt = account.provider().get_transaction_receipt(res.transaction_hash).await.unwrap();
+
+    sequencer.stop().expect("failed to stop sequencer");
+
+    match receipt {
+        MaybePendingTransactionReceipt::Receipt(TransactionReceipt::Declare(
+            DeclareTransactionReceipt { status, .. },
+        )) => {
+            assert_eq!(status, TransactionStatus::AcceptedOnL2);
+        }
+        _ => panic!("invalid tx receipt"),
+    }
+}
+
+fn prepare_contract_declaration_params(
```

Comment:
this is duplicated from migration [code](https://github.com/dojoengine/dojo/blob/24a4509f4db19c583569786f29cae76a0cfe5c60/crates/sozo/src/ops/migration/object.rs#L258), unsure where to place this. maybe we can have something like a `dojo-utils` crate?